preliz/distributions/truncatednormal.py

```diff
@@ -0,0 +1,415 @@
+                    * psi_alpha[0]
+                    - (psi_beta[1] ** 2 - 1)
+                    * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2))
+                    * psi_beta[0]
+                )
+                * (
+                    (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2)) * psi_alpha[0]
+                    - (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2)) * psi_beta[0]
+                )
+                / z**2
+            )
+            - (
+                3
+                * (
+                    (
+                        psi_alpha[1]
+                        * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2))
+                        * psi_alpha[0]
+                        - psi_beta[1]
+                        * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2))
+                        * psi_beta[0]
+                    )
+                    / z
+                )
+                ** 2
+            )
+            - (
+                6
+                * (
+                    (
+                        (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2)) * psi_alpha[0]
+                        - (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2)) * psi_beta[0]
+                    )
+                    / z
+                )
+                ** 4
+            )
+            + (
+                (psi_alpha[1] ** 3 - 3 * psi_alpha[1])
+                * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2))
+                * psi_alpha[0]
+                - (psi_beta[1] ** 3 - 3 * psi_beta[1])
+                * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2))
+                * psi_beta[0]
+            )
+            / z
+        )
+
+        denominator = (
+            1
+            + (
+                psi_alpha[1]
+                * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2))
+                * psi_alpha[0]
+                - psi_beta[1]
+                * (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2))
+                * psi_beta[0]
+            )
+            / z
+            - (
+                (
+                    (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_alpha[1] ** 2)) * psi_alpha[0]
+                    - (1 / (2 * np.pi) ** 0.5 * np.exp(-0.5 * psi_beta[1] ** 2)) * psi_beta[0]
+                )
+                / z
+            )
+            ** 2
+        ) ** 2
+
+        return numerator / denominator
+
+    def rvs(self, size=None, random_state=None):
+        random_state = np.random.default_rng(random_state)
+        random_samples = random_state.uniform(0, 1, size)
+        return nb_rvs(random_samples, self.mu, self.sigma, self.lower, self.upper)
+
+    def _fit_moments(self, mean, sigma):
+        # Assume gaussian
+        self._update(mean, sigma)
+
+    def _fit_mle(self, sample):
+        optimize_ml(self, sample)
```

Comment:
```suggestion
    def _fit_mle(self, sample):
        mean, sigma = mean_and_std(sample)
        self._update(mean, sigma, np.min(sample), np.max(sample))
        optimize_ml(self, sample)
```

Where `mean_and_std` is a function from preliz.internal.special

Reply:
This should fix the issue, actually setting the mean and sigma is not needed. This should also work with self._update(None, None, np.min(sample), np.max(sample))

Reply:
Thanks this seems to work...